source/fuzz/fuzzer_pass_donate_modules.cpp

```diff
@@ -1044,7 +1046,87 @@ void FuzzerPassDonateModules::AddLivesafeFunction(
       loop_limiter.set_increment_id(GetFuzzerContext()->GetFreshId());
       loop_limiter.set_compare_id(GetFuzzerContext()->GetFreshId());
       loop_limiter.set_logical_op_id(GetFuzzerContext()->GetFreshId());
-      loop_limiters.emplace_back(loop_limiter);
+
+      // We creating a branch from the back-edge block to the merge block. Thus,
+      // if merge block has any OpPhi instructions, we might need to adjust
+      // them.
+      //
+      // Note that the loop might not have a back-edge block, in which case
+      // we don't need to adjust anything.
```

Comment:
You mean: "the loop might already have an edge from the back-edge block to the merge block"

Reply:
No, that `if` statement handles the case when the back-edge block is unreachable in the CFG. Thus, the loop can't iterate and we don't need to adjust anything. The comment needs a refactoring, though. Fixed in bdc10e9.

Reply:
The case when loop already has an edge from the back-edge block to the merge block is handled below.

Reply:
Ah I see. I like the revised comment.